src/components/Select/Select/Select.tsx

```diff
@@ -55,18 +53,20 @@ export const Select = (props: TSingleSelectPropTypes): JSX.Element | null => {
     },
     labelAddons,
     tooltipAddons
-  } = props
-  const scrollRef = useRef<HTMLDivElement>(null)
-  const inputRef = useRef<HTMLInputElement | null>(null)
-  const { scrollHeight } = useGetElemSizes(scrollRef.current)
-  const [isOpen, setIsOpen] = useState<boolean>(false)
-  const containerRef = useRef<HTMLDivElement>(null)
-  const [searchValue, setSearchValue] = useState<string>('')
-  const [dropdownRef, setDropdownRef] = useState<HTMLDivElement | null>(null)
-  const currentSelection = (value as TItemValue) || selectedItem
-  const [selectedOption, setSelectedOption] = useState<TSelectOption | null>(null)
+  } = props;
 
-  const isWithSearch = withSearch && options.length > SELECTED_VISIBLE_MIN_COUNT
+  const scrollRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [dropdownRef, setDropdownRef] = useState<HTMLDivElement | null>(null);
+  const [isDynamicSearchEnabled, setIsDynamicSearchEnabled] = useState<boolean>(false);
+
+  const currentSelection = (value as TItemValue) || selectedItem;
+  const [selectedOption, setSelectedOption] = useState<TSelectOption | null>(null);
+
+  const isWithSearch = withSearch && (isDynamicSearchEnabled );
 
   const setCurrentSelectedLabel = useCallback(() => {
```

Comment:
unnecessary useCallback!

Reply:
@valodyapapikyan jan as i see `useCallback ` was needed due to we have dependency in useEffect  
``` 
useEffect(() => {
    setCurrentSelectedLabel()
  }, [setCurrentSelectedLabel])
  ```
  
  and if we will remove it the useEffect will be called each time